src/borg/archiver.py

```diff
@@ -1222,6 +1226,91 @@ def do_debug_dump_archive_items(self, args, repository, manifest, key):
         print('Done.')
         return EXIT_SUCCESS
 
+    def debug_json(self, d):
+        def decode_bytes(value):
+            # this should somehow be reversable later, but usual strings should
+            # look nice and chunk ids should mostly show in hex. Use a special
+            # inband signaling character (ASCII DEL) to distinguish between
+            # decoded and hex mode.
+            if not value.startswith(b'\x7f'):
+                try:
+                    value = value.decode()
+                    return value
+                except UnicodeDecodeError:
+                    pass
+            return '\u007f' + bin_to_hex(value)
+
+        def decode_tuple(t):
+            res = []
+            for value in t:
+                if isinstance(value, dict):
+                    value = decode(value)
+                elif isinstance(value, tuple) or isinstance(value, list):
+                    value = decode_tuple(value)
+                elif isinstance(value, bytes):
+                    value = decode_bytes(value)
+                res.append(value)
+            return res
+
+        def decode(d):
+            res = collections.OrderedDict()
+            for key, value in d.items():
+                if isinstance(value, dict):
+                    value = decode(value)
+                elif isinstance(value, tuple) or isinstance(value, list):
+                    value = decode_tuple(value)
+                elif isinstance(value, bytes):
+                    value = decode_bytes(value)
+                if isinstance(key, bytes):
+                    res[key.decode()] = value
+                else:
+                    res[key] = value
+            return res
+
+        return decode(d)
+
+    @with_repository()
+    def do_debug_dump_archive(self, args, repository, manifest, key):
+        """dump decoded archive metadata (not: data)"""
+
+        try:
+            archive_meta_orig = manifest.archives.get_raw_dict()[safe_encode(args.location.archive)]
```

Comment:
This is supposed to output the content of the data with as little alteration as needed to output as json. So i skipped as much higher level processing in borg as easily possible. Maybe i should do what i do in the manifest command and even load the manifest manually here. But that might not really be needed.